File: frontend/src/types/document.ts
import { Internship } from './internship'

export type MyDocument = {
  internshipId: number
  internship: Internship
  documentId: number
  documentType: DocumentType
  documentURL: string
}

export type MyDocumentType = {
  name: string
}

export enum DocumentTypeName {
  InternshipReport = 1,
  TurnitinReport = 2
}

File: backend/Internships/Internships.Application/Entities/Document.cs
namespace Internships.Core.Entities
{
    public class Document : AuditableBaseEntity
    {
        public int InternshipId { get; set; }
        public Internship Internship { get; set; }
        public int DocumentTypeId { get; set; }
        public DocumentType DocumentType { get; set; }

        public string DocumentURL { get; set; }
    }
}